src/components/atoms/Input/Input.test.js

```diff
@@ -0,0 +1,318 @@
+
+describe('Input components', () => {
+  let style
+  let name
+  let type
+  let icon
+  let label
+  let autoComplete
+  let autoFocus
+
+  let component
+  let props
+
+  beforeEach(() => {
+    style = { marginBottom: 0 }
+    name = 'input'
+    icon = 'shop'
+    label = 'Label'
+    autoComplete = 'off'
+    autoFocus = true
+  })
+
+  describe('Text', () => {
+    beforeEach(() => {
+      type = 'text'
+      props = {
+        style, name, type, icon, label, autoComplete, autoFocus
+      }
+      component = renderWithProviders(<Input
+        {...props}
+      />)
+    })
+
+    it('should match snapshot', () => {
+      const { asFragment } = component
+      expect(asFragment()).toMatchSnapshot()
+    })
+
+    it('should print correct label', () => {
+      const { container } = component
+      const span = container.querySelector('.input-label')
+
+      expect(span.textContent).toBe(label)
+    })
+
+    it('should allow letters to be inputted', () => {
+      const { container } = component
+      const elem = container.querySelector('#input')
+
+      fireEvent.change(elem, { target: { value: 'abc23!@' } })
+      expect(elem.value).toBe('abc23!@')
+    })
+
+    it('should allow letters to be deleted', () => {
+      const { container } = component
+      const elem = container.querySelector('#input')
+
+      fireEvent.change(elem, { target: { value: 'abc23!@' } })
+      fireEvent.change(elem, { target: { value: 'abc2' } })
+      expect(elem.value).toBe('abc2')
+    })
+  })
+
+  describe('Email', () => {
+    beforeEach(() => {
+      type = 'email'
+      props = {
+        style, name, type, icon, label, autoComplete, autoFocus
+      }
+      component = renderWithProviders(<Input
+        {...props}
+      />)
+    })
+
+    it('should match snapshot', () => {
+      const { asFragment } = component
+      expect(asFragment()).toMatchSnapshot()
+    })
+
+    it('should print correct label', () => {
+      const { container } = component
+      const span = container.querySelector('.input-label')
+
+      expect(span.textContent).toBe(label)
+    })
+
+    it('should contain icon', () => {
+      const { container } = component
+      const elem = container.querySelector('.input-icon svg')
+      expect(elem).toBeTruthy()
+    })
+
+    it('should allow letters to be inputted', () => {
+      const { container } = component
+      const elem = container.querySelector('#input')
+
+      fireEvent.change(elem, { target: { value: 'id@mailservice.com' } })
+      expect(elem.value).toBe('id@mailservice.com')
+    })
+
+    it('should allow letters to be deleted', () => {
+      const { container } = component
+      const elem = container.querySelector('#input')
+
+      fireEvent.change(elem, { target: { value: 'id@mailservice.com' } })
+      fireEvent.change(elem, { target: { value: 'id@mailser' } })
+      expect(elem.value).toBe('id@mailser')
+    })
```

Comment:
the email input is just a `TextInput` with an additional  pattern props:

```
          <TextInput
            {...props}
            pattern={pattern || /^(([^<>()[\]\\.,;:\s@']+(\.[^<>()[\]\\.,;:\s@']+)*)|('.+'))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/}
          >
            {children}
          </TextInput>
```

... so this test cases are not needed.